tests/lib/hardware/hardware_base.py

```diff
@@ -151,13 +151,11 @@ def ansible_run_playbook(self, playbook: str,
         else:
             limit = ""
 
-        # Supplied extra_vars from settings always take precedence
-        if settings.ANSIBLE_EXTRA_VARS:
-            extra_vars.update(json.loads(settings.ANSIBLE_EXTRA_VARS))
+        extra_vars_param = ""
         if extra_vars:
-            extra_vars_param = f"--extra-vars '{json.dumps(extra_vars)}'"
-        else:
-            extra_vars_param = ""
+            extra_vars_param += f" --extra-vars '{json.dumps(extra_vars)}'"
```

Comment:
We probably shouldn't json.dumps here, and instead require the caller to decide how the param is formatted.

That can be a follow up though.

Reply:
Agreed. Will do a follow up PR

Reply:
The *ansible_run_playbook* function signature defaults extra_vars to dict, could you please update the function inline  where you describe parameters and acceptable value formats.

Reply:
Lets update the function signature when we expand the use of the param to be any input into --extra-vars.